docs/pipelineruns.md

```diff
@@ -627,6 +627,20 @@ Task Runs:
         foo
 ```
 
+The name of the `TaskRuns` and `Runs` owned by a `PipelineRun`  are univocally associated to the owning resource.
+If a `PipelineRun` resource is deleted and created with the same name, the child `TaskRuns` will be created with the
+same name as before. The base format of the name is `<pipelinerun-name>-<pipelinetask-name>`. The name may vary
+according the logic of [`kmeta.ChildName`](https://pkg.go.dev/github.com/knative/pkg/kmeta#ChildName).
+
+Some examples:
+
+| `PipelineRun` Name       | `PipelineTask` Name          | `TaskRun` Name    |
+|--------------------------|------------------------------|-------------------|
+| pipeline-run             | task1                        | my-pipeline-task1 |
+| pipeline-run             | task2-0123456789-0123456789-0123456789-0123456789-0123456789 | pipeline-runee4a397d6eab67777d4e6f9991cd19e6-task2-0123456789-0 |
```

Comment:
how are the dashes added when one or both of the resource names are long?

No dash after the `pipelineRun` name here and in the last example.

`pipeline-runee4a397d6eab67777d4e6f9991cd19e6-task2-0123456789-0` (pipelineRun name is `pipeline-run` and pipelineTask name is long)
`pipeline-run-0123456789-012345607ad8c7aac5873cdfabe472a68996b5c` (pipelineRun name and pipelineTask name are long)

Reply:
`ChildName` does not add a dashes.
The way we're using it is to prepend a dash to the `PipelineTask` name.

```golang
     return kmeta.ChildName(prName, fmt.Sprintf("-%s", ptName))
```

When both names are too long an hash of the whole thing is used so that specific dash may not show up in the final name.